.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,44 @@
+---
+name: golangci-lint
+
+on:
+  pull_request:
+    branches: ['*']
+
+jobs:
+  golangci-lint:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set GO_VERSION
+        run: echo "GO_VERSION=$(grep -m 1 -r 'go' go.mod | awk '{print $2}')" >> $GITHUB_ENV
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          version: v1.45.2
```

Comment:
1.46.2 is the latest release.

Reply:
Okay, I think I can point it to latest one using `latest`
Something like this
```
        with:
          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
          version: latest
```

Reply:
Yes we can do it but the only problem is you might need to fix linter issues if a new linter is added to the latest release of golangci-lint, it can block PRs from getting merged. if it's okay we can point to the latest.

Reply:
Ohh understood! I think we should stick to a specific version, if required we can update the version.

Reply:
Can we resolve this?

Reply:
@Madhu-1 Please resolve if you are satisfied

Reply:
i dont see any option to resolve it. approved the changes.